src/controllers/v1/members.ts

```diff
@@ -59,19 +75,36 @@ export class MembersController extends Controller {
     if (!nodeMembers.includes(aliasOrAddress)) {
       throw new NotFound('Member does not exist')
     }
+    if (aliasOrAddress === this.env.get('SELF_ADDRESS')) {
+      return {
+        alias: aliasOrAddress,
+        address: aliasOrAddress,
+        role: 'Self',
+      }
+    }
 
     return {
       alias: aliasOrAddress,
       address: aliasOrAddress,
+      role: role,
     }
   }
 
   @SuccessResponse(200)
   @Response<NotFound>(404, 'Not found')
   @Response<Conflict>(409, 'Conflict')
-  @Put('/{address}')
-  public async put(@Path('address') address: Address, @Body() body: { alias: Alias }): Promise<Member> {
+  @Patch('/{address}')
+  public async patch(@Path('address') address: Address, @Body() body: MemberBody): Promise<Member> {
     const { alias } = body
+    let role: RoleRow | undefined = undefined
+    if (body.role) {
+      const rolesFromDb = await this.db.get('roles')
+      if (!isValidRole(rolesFromDb, body.role)) {
+        throw new BadRequest('Invalid role')
+      }
+      const [roleFromDb] = await this.db.get('roles', { role: body.role })
+      role = roleFromDb
```

Comment:
There shouldn't be a need for two requests here. Just do the second query and if it's not returned it's not valid